### PR TITLE
Disable redirects

### PR DESCRIPTION
**Pull request by mdellweg**
Also change default url to https://localhost

Fixes #7666
https://pulp.plan.io/issues/7666

### CODE REVIEW

**Comment by mdellweg** [comment]
@bmbouter can you test this? and also with --base-url "http://localhost"?

I don't know, yet, if redirecting is really disabled. Also interesting is whether it produces a better error message.

**Comment by bmbouter** [comment]
I can confirm that defaulting to https resolves the issue:

```
(pulp) [vagrant@pulp3-source-fedora32 pulpcore]$ pulp --no-verify-ssl file repository create --name file_repo2
/usr/local/lib/pulp/lib64/python3.8/site-packages/urllib3/connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
/usr/local/lib/pulp/lib64/python3.8/site-packages/urllib3/connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
{
  "pulp_href": "/pulp/api/v3/repositories/file/file/a545fc8e-af5f-455b-93d6-690b46017b90/",
  "pulp_created": "2020-10-08T15:46:00.862551Z",
  "versions_href": "/pulp/api/v3/repositories/file/file/a545fc8e-af5f-455b-93d6-690b46017b90/versions/",
  "latest_version_href": "/pulp/api/v3/repositories/file/file/a545fc8e-af5f-455b-93d6-690b46017b90/versions/0/",
  "name": "file_repo2",
  "description": null,
  "remote": null
}
```

**Comment by bmbouter** [comment]
I can also confirm that it also blocks redirection:

```
(pulp) [vagrant@pulp3-source-fedora32 pulpcore]$ pulp  --base-url "http://localhost" --no-verify-ssl file repository create --name file_repo2
Traceback (most recent call last):
  File "/home/vagrant/devel/pulp-cli/pulpcore/cli/openapi.py", line 49, in load_api
    raise IOError()
OSError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/pulp/bin/pulp", line 11, in <module>
    load_entry_point('pulp-cli', 'console_scripts', 'pulp')()
  File "/usr/local/lib/pulp/lib64/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/pulp/lib64/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/pulp/lib64/python3.8/site-packages/click/core.py", line 1256, in invoke
    Command.invoke(self, ctx)
  File "/usr/local/lib/pulp/lib64/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/pulp/lib64/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/pulp/lib64/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/vagrant/devel/pulp-cli/pulpcore/cli/common.py", line 63, in main
    ctx.obj.api = OpenAPI(
  File "/home/vagrant/devel/pulp-cli/pulpcore/cli/openapi.py", line 36, in __init__
    self.load_api(refresh_cache=refresh_cache)
  File "/home/vagrant/devel/pulp-cli/pulpcore/cli/openapi.py", line 55, in load_api
    data = self._download_api()
  File "/home/vagrant/devel/pulp-cli/pulpcore/cli/openapi.py", line 82, in _download_api
    raise Exception("Attempt to redirect blocked.")
Exception: Attempt to redirect blocked.
```

**Comment by bmbouter** [comment]
I think this SO explains why the redirection is changing to GET ... https://softwareengineering.stackexchange.com/questions/99894/why-doesnt-http-have-post-redirect

**Review by daviddavis — APPROVED** [review]

**Comment by mdellweg** [comment]
@daviddavis redirecting is difficult... we seem to redirect in some places of the api.
I'm tempted to just set the default to https://localhost and document, that redirecting from http does not work.

**Comment by bmbouter** [comment]
> @daviddavis redirecting is difficult... we seem to redirect in some places of the api.
> I'm tempted to just set the default to https://localhost and document, that redirecting from http does not work.

+1 to changing the default and adding some docs. I think the specific situation we struggle with is when a POST or PUT request receives a redirect. This is due to Pulp's server using 301 and the requests library (like most) does not follow the redirect AND keep the method type, it switches it to GET.

I think some explanation about why would be helpful, instead of saying it doesn't work without an explanation.

File: README.md [hunk 1]
```diff
@@ -12,6 +12,13 @@ options include:
   * `--user`
   * `--password`
 
+## Known issues
+
+  * Redirecting from `http` to `https`, as done by a typical pulp installation,
```
Comment (bmbouter) [inline README.md:17]: nitpick 

```suggestion
  * Redirecting from `http` to `https`, as done by a typical Pulp installation,
```

**Pull request merged**